brilirs/src/interp.rs

```diff
@@ -516,19 +622,58 @@ fn execute_value_op<T: std::io::Write>(
       let res = Value::Pointer(arg0.add(arg1));
       state.env.set(dest, res);
     }
+    Resolve => {
+      let res = match handlers.remove(&args[0]) {
+        Some(handle) => handle.0.join().unwrap(),
+        None => panic!("Unexpected execution error"),
+      };
+
+      state.env.set(dest, res.unwrap());
+    }
+    CompareAndSwap => {
+      //atomic
+      let arg0 = get_arg::<usize>(&state.env, 0, args);
+      //expected
+      let arg1 = get_arg::<i64>(&state.env, 1, args);
+      //update
+      let arg2 = get_arg::<i64>(&state.env, 2, args);
+      let res = state.atomics.compare_and_swap(arg0, arg1, arg2);
+      state.env.set(dest, Value::Int(res));
+    }
+    LoadAtomic => {
+      let arg0 = get_arg::<usize>(&state.env, 0, args) as usize;
```

Comment:
Unnecessary `as usize` cast. I think clippy has some suggestions that could be applied.